core/src/services/dropbox/response.rs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use serde::Deserialize;
+
+#[derive(Default, Debug, Deserialize)]
+#[serde(default)]
+pub struct DropboxErrorResponse {
+    pub error_summary: String,
+    pub error: DropboxErrorDetail,
+}
+
+#[derive(Default, Debug, Deserialize)]
+#[serde(default)]
+pub struct DropboxErrorDetail {
+    #[serde(rename(deserialize = ".tag"))]
+    pub tag: String,
+}
+
+#[derive(Default, Debug, Deserialize)]
+#[serde(default)]
+pub struct DropboxMetadataResponse {
```

Comment:
What about making those Option<T> options a separate struct and flatten them in serialization.

Reply:
Hi, I have some ideas about PR review in OpenDAL. We should focus on the job that the current PR is working on, which is to add stat support.

The implementation details can be improved later. It's important not to extend the scope of this PR and instead concentrate on **required** changes rather than **good-to-have** ones. This will make our contributing experience much better.

If you believe this change is important, you can submit a new pull request to improve it after this one has been merged.

Reply:
- **required** changes: API semantic, comments, error handling...
- **good to have** changes: internal struct layout, const extraction...